iothub_client/inc/iothub_message.h

```diff
@@ -191,9 +191,9 @@ MOCKABLE_FUNCTION(, MAP_HANDLE, IoTHubMessage_Properties, IOTHUB_MESSAGE_HANDLE,
 *
 * @param   iotHubMessageHandle Handle to the message.
 *
-* @param   key name of the property to set.  Note that when sending messages via the HTTP transport, this value must not contain spaces.
+* @param   key name of the property to set using valid ASCII characters.  Note that when sending messages via the HTTP transport, this value must not contain spaces.
```

Comment:
this is not true, it's defined as below

Property names and values can only contain ASCII **alphanumeric** characters, plus {'!', '#', '$', '%, '&', ''', '*', '+', '-', '.', '^', '_', '`', '|', '~'} 

https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct

Reply:
How do you think I best explain that here?  I was hoping "valid" would suffice.  I could just say between dec value [32, 126]

Reply:
or [' ', '~']

Reply:
Except this page says there should be no spaces.... https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/iothub-message-h/iothubmessage-setproperty

Reply:
a space is not an ASCII alphanumeric character

Reply:
The docs specify no space for the key.  But no mention for the value.
The code allows spaces.  Are you saying we should not allow spaces for the value either?